build.gradle

```diff
@@ -1,66 +1,112 @@
+    ]
+
+    servicePackage = 'org.openbroker.cloudevents'
 }
 
-apply from: "jar.gradle"
+sourceCompatibility = jdkVersion
+targetCompatibility = jdkVersion
 
 repositories {
+    maven {
+        url "https://jitpack.io"
+        content {
+            includeGroupByRegex 'com\\.(github|gitlab|gitee).*'
+        }
+    }
     jcenter()
     mavenCentral()
-    maven { url "https://dl.bintray.com/kotlin/kotlinx" }
 }
 
 dependencies {
-    // Kotlin
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
 
-    // Jackson
-    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-core', version:'2.9.8'
-    implementation group: 'com.fasterxml.jackson.module', name: 'jackson-module-kotlin', version:'2.9.8'
-    implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version:'2.9.8'
+    implementation(
+        platform("com.fasterxml.jackson:jackson-bom:$ver.jackson"),
+        platform("org.jetbrains.kotlin:kotlin-bom:$ver.kotlin"),
+        "org.jetbrains.kotlin:kotlin-stdlib-jdk8",
+        "com.fasterxml.jackson.core:jackson-core",
+        "com.fasterxml.jackson.module:jackson-module-kotlin",
+        "com.fasterxml.jackson.datatype:jackson-datatype-jsr310",
+    )
 
-    testImplementation("org.junit.jupiter:junit-jupiter-api:5.4.0")
-    testRuntime("org.junit.jupiter:junit-jupiter-engine:5.4.0")
+    testImplementation(
+        platform("org.junit:junit-bom:$ver.junit"),
+        "org.junit.jupiter:junit-jupiter-api"
+    )
+    testRuntime(
+        "org.junit.jupiter:junit-jupiter-engine"
+    )
 }
 
 tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
     kotlinOptions {
-        languageVersion = kotlin_api_version
-        apiVersion = kotlin_api_version
-        jvmTarget = jvm_version
-        javaParameters = true   // Useful for reflection.
+        javaParameters = true
+        jvmTarget = jdkVersion
+        freeCompilerArgs += ['-Xjsr305=strict', '-progressive']
+        languageVersion = kotlinLanguageVersion
+        apiVersion = kotlinLanguageVersion
     }
 }
 
-compileJava {
-    sourceCompatibility = jvm_version
-    targetCompatibility = jvm_version
-    options.incremental = true
-    options.encoding = 'UTF-8'
+tasks.withType(JavaCompile).all {
+    options.encoding = defaultEncoding
+    options.compilerArgs = ['-parameters', '-Xlint:unchecked']
+}
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    archiveClassifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            from components.java
+            artifact sourcesJar
+        }
+    }
 }
 
 test {
+
     useJUnitPlatform()
-}
 
-wrapper {
-    description = 'Generates gradlew[.bat] scripts for faster execution'
-    gradleVersion = '5.2.1'
+    reports.junitXml.enabled false
+    reports.html.enabled false
+
+    failFast = System.getProperty('failFast') != null
```

Comment:
We don't want the test to fail fast. We want to see all failed test so we can act on them. Our test suit take just just 1-2 sec to run.